### PR TITLE
feat(bidsificator) : new app

### PR DESCRIPTION
**Pull request by floriansipp**
New app, some env and poetry magic because pyqt is not working with python3.12 yet

### CODE REVIEW

File: apps/bidsificator/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    libxcb-keysyms1 libglib2.0-0 libdbus-1-3 libxcb-icccm4 \
+    libxcb-randr0 libxcb-render-util0 libxcb-shape0 fontconfig \
+    libfontconfig1 libxcb-image0 libxcb-util1 && \
+    git clone https://github.com/HIP-infrastructure/Bidsificator.git && \
+    # setup python environment and poetry \
+    python3.10 -m venv .venv && \
+    .venv/bin/pip install poetry && \
+    .venv/bin/poetry config virtualenvs.path /apps/${APP_NAME}/.venv && \
+    .venv/bin/poetry config virtualenvs.in-project true && \
+    .venv/bin/poetry -C Bidsificator install && \
```
Comment (greut) [inline apps/bidsificator/Dockerfile:32]: a poetry project can be installed using `pip install`.

File: apps/bidsificator/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    apt-get update -qy && \
+    apt-get install --no-install-recommends -qy \
+    git python3.10 python3.10-venv python3.10-dev python3-pip \
+    # lib necessary to have a working pyqt application \
+    libxcb-icccm4 libxcb-randr0 libxcb-render-util0 \
+    libxcb-shape0 libxcursor1 libxkbcommon-x11-0 \
+    libxcb-keysyms1 libglib2.0-0 libdbus-1-3 libxcb-icccm4 \
+    libxcb-randr0 libxcb-render-util0 libxcb-shape0 fontconfig \
+    libfontconfig1 libxcb-image0 libxcb-util1 && \
+    git clone https://github.com/HIP-infrastructure/Bidsificator.git && \
```
Comment (greut) [inline apps/bidsificator/Dockerfile:26]: `ADD` (or `COPY`) supports git projects as well, or their archive via HTTP.

File: apps/bidsificator/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    # setup python environment and poetry \
+    python3.10 -m venv .venv && \
```
Comment (greut) [inline apps/bidsificator/Dockerfile:28]: can you fix _bidsificator_ to support whatever version the base image is using?

File: apps/bidsificator/build.sh [hunk 4]
```diff
@@ -0,0 +1,31 @@
+#!/bin/sh
+
+APP_NAME="bidsificator"
+APP_VERSION="1.0.0"
```
Comment (greut) [inline apps/bidsificator/build.sh:4]: is it used?

File: apps/bidsificator/Dockerfile (hunk 3 above)
Comment (floriansipp) [reply apps/bidsificator/Dockerfile:28]: Nop, it depends on pyqt and some dependency hell. 
The only way would be to rewrite the soft in c++ (maybe with something that packages python into executable too but didn't have time to investigate that yet and there are more pressing things to do before imo)

File: apps/bidsificator/Dockerfile (hunk 2 above)
Comment (floriansipp) [reply apps/bidsificator/Dockerfile:26]: Good to know 🫡

File: apps/bidsificator/Dockerfile (hunk 3 above)
Comment (greut) [reply apps/bidsificator/Dockerfile:28]: Ok, then downgrade the base ubuntu to Jammy, which is still on Python 3.10.

BTW, we really need to stop bundling those binary files: https://github.com/HIP-infrastructure/Bidsificator/tree/main/bidsificator/core/PyEEGFormat

File: apps/bidsificator/build.sh (hunk 4 above)
Comment (greut) [reply apps/bidsificator/build.sh:4]: you don't have to package it, only to tag it.

Comment (floriansipp) [reply apps/bidsificator/build.sh:4]: That i can do

File: apps/bidsificator/Dockerfile (hunk 3 above)
Comment (floriansipp) [reply apps/bidsificator/Dockerfile]: That will be done at a later date, i still didn't take time to contact Lyon's PI to deal with that and put the code in public

File: apps/bidsificator/build.sh [hunk 5]
```diff
@@ -0,0 +1,31 @@
+#!/bin/sh
+
```
Comment (greut) [inline apps/bidsificator/build.sh:2]:
```suggestion

set -e

```

File: apps/bidsificator/build.sh [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    -t ${REGISTRY}/${APP_NAME}:${VERSION} \
+    --label "APP_NAME=${APP_NAME}" \
+    --label "APP_VERSION=${APP_VERSION}" \
+    --build-arg "APP_NAME=${APP_NAME}" \
+    --build-arg "APP_VERSION=${APP_VERSION}" \
+    --no-cache \
+    --output=$OUTPUT \
+    .
+
+rm -rf core
```
Comment (greut) [inline apps/bidsificator/build.sh:31]:
```suggestion
cp -r ../../core ./core
trap "rm -rf core" EXIT

docker buildx build \
    --pull \
    -t ${REGISTRY}/${APP_NAME} \
    -t ${REGISTRY}/${APP_NAME}:${VERSION} \
    --label "APP_NAME=${APP_NAME}" \
    --label "APP_VERSION=${APP_VERSION}" \
    --build-arg "APP_NAME=${APP_NAME}" \
    --build-arg "APP_VERSION=${APP_VERSION}" \
    --no-cache \
    --output=$OUTPUT \
    .
```

File: apps/bidsificator/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+    libxcb-shape0 libxcursor1 libxkbcommon-x11-0 \
+    libxcb-keysyms1 libglib2.0-0 libdbus-1-3 libxcb-icccm4 \
+    libxcb-randr0 libxcb-render-util0 libxcb-shape0 fontconfig \
+    libfontconfig1 libxcb-image0 libxcb-util1 && \
+    # unpack and put the app in the right place \ 
+    unzip V${APP_VERSION}.zip && \
+    mv Bidsificator-${APP_VERSION}/* . && \
+    rm -rf Bidsificator-${APP_VERSION} V${APP_VERSION}.zip && \
+    # install poetry project via pip \
+    pip install . && \
```
Comment (greut) [inline apps/bidsificator/Dockerfile:28]: Maybe you'll need git back again.

```bash
pip install --no-cache git+https://github.com/HIP-infrastructure/Bidsificator/@v${APP_VERSION}#egg=Bidsificator
```

Comment (floriansipp) [reply apps/bidsificator/Dockerfile]: with an uppercase V, but it works nicely

Comment (greut) [reply apps/bidsificator/Dockerfile]: Big V? who does that?

Comment (floriansipp) [reply apps/bidsificator/Dockerfile]: image

File: apps/bidsificator/Dockerfile [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+    --mount=type=cache,target=/var/lib/apt,sharing=locked \
+    apt-get update -qy && \
+    apt-get install --no-install-recommends -qy \
+    git python3 python3-pip \
+    # lib necessary to have a working pyqt application \
+    libxcb-icccm4 libxcb-randr0 libxcb-render-util0 \
+    libxcb-shape0 libxcursor1 libxkbcommon-x11-0 \
+    libxcb-keysyms1 libglib2.0-0 libdbus-1-3 libxcb-icccm4 \
+    libxcb-randr0 libxcb-render-util0 libxcb-shape0 fontconfig \
+    libfontconfig1 libxcb-image0 libxcb-util1 && \
```
Comment (greut) [inline apps/bidsificator/Dockerfile:21]: you've got some duplicates in here.

File: apps/bidsificator/Dockerfile [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+    # install using pip and git \
+    pip install --no-cache git+https://github.com/HIP-infrastructure/Bidsificator/@V${APP_VERSION}#egg=Bidsificator && \
```
Comment (greut) [inline apps/bidsificator/Dockerfile:23]: https://github.com/HIP-infrastructure/Bidsificator/blob/V1.0.0/pyproject.toml#L3 :wink:

Comment (floriansipp) [reply apps/bidsificator/Dockerfile:23]: oh i forgot this existed ahah

Comment (floriansipp) [reply apps/bidsificator/Dockerfile]: okidoki this was updated and i switched to lowercase v for version numbering

**Review by iDmple — APPROVED** [review]

**Pull request merged**